core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLParams.scala

```diff
@@ -0,0 +1,109 @@
+package com.microsoft.azure.synapse.ml.causal
+
+import com.microsoft.azure.synapse.ml.core.contracts.{HasFeaturesCol, HasWeightCol}
+import org.apache.spark.ml.classification.{LogisticRegression, ProbabilisticClassifier}
+import org.apache.spark.ml.{Estimator, Model}
+import com.microsoft.azure.synapse.ml.param.{EstimatorParam, UntypedArrayParam}
+import org.apache.spark.ml.ParamInjections.HasParallelismInjected
+import org.apache.spark.ml.param.shared.HasMaxIter
+import org.apache.spark.ml.param.{IntParam, Param, Params}
+import org.apache.spark.ml.regression.Regressor
+
+trait HasTreatmentCol extends Params {
+  val treatmentCol = new Param[String](this, "treatmentCol", "treatment column")
+  def getTreatmentCol: String = $(treatmentCol)
+
+  /**
+   * Set name of the column which will be used as treatment
+   *
+   * @group setParam
+   */
+  def setTreatmentCol(value: String): this.type = set(treatmentCol, value)
+}
+
+trait HasOutcomeCol extends Params {
+  val outcomeCol: Param[String] = new Param[String](this, "outcomeCol", "outcome column")
+  def getOutcomeCol: String = $(outcomeCol)
+
+  /**
+   * Set name of the column which will be used as outcome
+   *
+   * @group setParam
+   */
+  def setOutcomeCol(value: String): this.type = set(outcomeCol, value)
+}
+
+trait LinearDMLParams extends Params
+  with HasTreatmentCol with HasOutcomeCol with HasFeaturesCol
+  with HasMaxIter with HasWeightCol with HasParallelismInjected {
+
+  val treatmentModel = new EstimatorParam(this, "treatmentModel", "treatment model to run")
+  def getTreatmentModel: Estimator[_ <: Model[_]] = $(treatmentModel)
+
+  /**
+   * Set treatment model, it could be any model derived from 'org.apache.spark.ml.regression.Regressor' or 'org.apache.spark.ml.classification.ProbabilisticClassifier'
+   *
+   * @group setParam
+   */
+  def setTreatmentModel(value: Estimator[_ <: Model[_]]): this.type = {
+    val isSupportedModel = value match {
+      case regressor: Regressor[_,_,_] =>   true // for continuous treatment
+      case classifier: ProbabilisticClassifier[_, _, _] => true
+      case _ => false
+    }
```

Comment:
Duplicated code with `setOutcomeModel` function below. Consider extract this part of the logic to a function.